### DISCUSSION TITLE
Passwordless Authentication

### DISCUSSION THREAD

**Discussion by assman** [fastapi/fastapi#6652] [Questions] [answered]
Hi

Would like to seek some advice on how should I go about implementing a passwordless flow using FastAPI. I read through the FastAPI docs that mentioned the password flow using FastAPIs **OAuth2PasswordBearer**. Would like to use a token or JWT but with a passwordless flow.

The passwordless flow would be an email/sms based flow where the user would get an OTP or a login link via email/sms.

**Comment by ArcLightSlavik** [comment]
I suppose you mean logging via google/facebook?
https://medium.com/data-rebels/fastapi-google-as-an-external-authentication-provider-3a527672cf33

**Comment by assman** [comment]
Sorry my bad, wasn't clear enough in my description of the issue, I have updated it.

**Comment by Hultner** [comment, marked as answer]
Well you could create a JWT-token with short expire time (and nbf/not before to time of generation), but enough for the email/sms to be received. I'd probably go somewhere around 30 minutes.

Another possibility is to offload auth entirely and use an external IDP. I recently did this for a project where I use Keycloak via OpenID Connect Authorisation Code Flow. Keycloak in its turn got built in support to add arbitrary OAuth, SAML, Active Directory, LDAP integrations and preconfigured templates for common social logins like GitHub, Microsoft etc.

**Comment by assman** [comment]
Ok thank you for sharing

**Comment by Garito** [comment]
For real passwordless app use metamask and message signature for critical operations

**Comment by bert2002** [comment]
@Garito did you manage to get a metamask integration running?
